### DISCUSSION TITLE
0.13.0

### DISCUSSION THREAD

**Discussion by fredizzimo** [neovide/neovide#2552] [General]
## What's Changed
The main highlights are:
* Improved and more configurable font rendering, see `neovide_text_contrast` and `neovide_text_gamma`
* D3D for smoother rendering on Windows
* Fully working smooth scrolling in Windows with borders and highlights
* Better stability compared to 0.12.x

### Breaking changes
* Neovim 0.10.0 is now required
* The font size in `config.toml` is now specified in points instead of pixels, to match the behaviour of guifont and most other software
* The macOS option `neovide_input_macos_alt_is_meta` has been renamed to neovide_input_macos_option_key_is_meta and instead of `true/false` it can now be set to `only_left`, `only_right`, `both` or `none`
* The font sizes are now slightly different and matches that set in most other programs due to the fractional font support. 
* The `--fork` option has been reversed, and `--no-fork` is now the default. If the old behaviour is desired, you can set the environment variable `NEOVIDE_FORK=1` 
* Closing the Window when using remote connections now also exits Neovim. If you want to keep the remote instance running, you can detach using `:call chanclose(g:neovide_channel_id)`

### Known issues
* Neovide will hang on some Windows systems. This has turned out to be a very tricky bug to track down, since it only occurs on a few systems. See https://github.com/neovide/neovide/pull/2463 for more information. Any help debugging this is highly appreciated.

### Features

* fix: font width can't be configured with config file by @gmr458 in https://github.com/neovide/neovide/pull/2331
* feat: add support for hiding window title from config file by @falcucci in https://github.com/neovide/neovide/pull/2344
* doc: instructions of profiling with tracy. by @crupest in https://github.com/neovide/neovide/pull/2356
* feat(macos): display border for opaque backgrounds by @emonadeo in https://github.com/neovide/neovide/pull/2351
* feat: add window menu on mac by @polachok in https://github.com/neovide/neovide/pull/2323
* feat(macos): drop multiple files and folders on macOS by @falcucci in https://github.com/neovide/neovide/pull/2396
* feat: Smooth cursor blink animation option by @agraven in https://github.com/neovide/neovide/pull/2421
* feat: tabs option as ENV variable and config file position + prefix cleanup for CLI argument by @nazriel in https://github.com/neovide/neovide/pull/2441
* feat: Apply winit window blur setting on all platforms by @agraven in https://github.com/neovide/neovide/pull/2440
* feat(windows): Direct3D Rendering on Windows by @fredizzimo in https://github.com/neovide/neovide/pull/2215
* feat: add ability to open files from Finder in macOS by @polachok in https://github.com/neovide/neovide/pull/2395
* feat: render same z-index together by @Kethku in https://github.com/neovide/neovide/pull/2467
* feat!: Option key as meta configuration by @9mm in https://github.com/neovide/neovide/pull/2486
* feat!: Support fractional grid and font sizes by @fredizzimo in https://github.com/neovide/neovide/pull/2485, https://github.com/neovide/neovide/pull/2500
* feat: Configurable contrast and gamma by @fredizzimo in https://github.com/neovide/neovide/pull/2510
* fix: relative WSL paths and paths with spaces by @fredizzimo in https://github.com/neovide/neovide/pull/2507

### Bug fixes
* fix: Handle some more numpad keys by @hhirtz in https://github.com/neovide/neovide/pull/2334
* Update commands.md missing link by @dineshKumar777 in https://github.com/neovide/neovide/pull/2341
* fix(macos): bug of restoring window position on macos. by @crupest in https://github.com/neovide/neovide/pull/2345
* fix: loading of fonts with OpenType font variations by @fredizzimo in https://github.com/neovide/neovide/pull/2354
* fix(wsl): #2049 ignore lines matching wsl screen size error  by @DZappala in https://github.com/neovide/neovide/pull/2374
* docs: fix show_border cutting into transparent by @emonadeo in https://github.com/neovide/neovide/pull/2380
* fix(macos): execute login before starting login shell by @falcucci in https://github.com/neovide/neovide/pull/2405
* fix: Handle new viewport margins event (fixes smooth scrolling with borders and winbars) by @Kethku in https://github.com/neovide/neovide/pull/2455
* fix: Prevent intro message hack from showing when the intro message is fixed upstream by @Kethku in https://github.com/neovide/neovide/pull/2460
* fix: Convert Windows paths to WSL when launching in WSL mode by @Holzhaus in https://github.com/neovide/neovide/pull/2256
* fix: Determine the window under the mouse on the fly by @fredizzimo in https://github.com/neovide/neovide/pull/2311
* fix: save grid size for maximized windows by @sid-6581 in https://github.com/neovide/neovide/pull/2528
* fix:! Don't fork by default by @fredizzimo in https://github.com/neovide/neovide/pull/2512
* fix!: Perform a clean exit again by @fredizzimo in https://github.com/neovide/neovide/pull/2463
* fix!: Use points instead of pixels for the config file by @fredizzimo in https://github.com/neovide/neovide/pull/2476
* fix(docs): fix reversed autocmds for setting input_ime by @cpmsmith in https://github.com/neovide/neovide/pull/2545

### Other changes

 (including fixes for bugs that were not part of the previous release)
* chore: update lru crate to fix building with the nightly toolchain by @fredizzimo in https://github.com/neovide/neovide/pull/2346
* Update nvim-rs, remove superflous arc-wrapping by @KillTheMule in https://github.com/neovide/neovide/pull/2336
* chore: remove legacy objc and cocoa by @polachok in https://github.com/neovide/neovide/pull/2390
* chore: Upgrade dependencies and remove unused by @agraven in https://github.com/neovide/neovide/pull/2443
* chore: remove swash's scale and render features by @tim-harding in https://github.com/neovide/neovide/pull/2450
* chore: Support multiple render backends by @fredizzimo in https://github.com/neovide/neovide/pull/2359
* fix: set startup directory failed is not an error by @TwIStOy in https://github.com/neovide/neovide/pull/2457
* chore(CI): Enable full build on pull requests by @fredizzimo in https://github.com/neovide/neovide/pull/2473
* chore: Refactor the update loop by @fredizzimo in https://github.com/neovide/neovide/pull/2470
* fix: Temporary hang after sleep by @fredizzimo in https://github.com/neovide/neovide/pull/2472
* chore: Fix clippy warnings by @fredizzimo in https://github.com/neovide/neovide/pull/2483
* feat: draw block characters with background opacity (was reverted) by @Theaninova in https://github.com/neovide/neovide/pull/2478
* chore: Use typesafe pixel and grid units by @fredizzimo in https://github.com/neovide/neovide/pull/2484
* fix: address viewport margin regression by sorting DrawCommand batches by @Kethku in https://github.com/neovide/neovide/pull/2487
* chore: Refactor the registry-relevant functions. by @InfyniteHeap in https://github.com/neovide/neovide/pull/2490
* fix: clipping rect regression by @fredizzimo in https://github.com/neovide/neovide/pull/2494
* Revert "feat!: Support fractional grid and font sizes" by @Kethku in https://github.com/neovide/neovide/pull/2497
* chore: migrate dependency from `winapi` to `windows`. by @InfyniteHeap in https://github.com/neovide/neovide/pull/2505
* fix: Cursor offset by @fredizzimo in https://github.com/neovide/neovide/pull/2511
* fix: respect floating shadow setting in layer rendering by @sid-6581 in https://github.com/neovide/neovide/pull/2526
* fix: fix the CI by @fredizzimo in https://github.com/neovide/neovide/pull/2534
* chore: yet another fix for the CI by @fredizzimo in https://github.com/neovide/neovide/pull/2535
* fix: don't group floating windows unless one is fully contained in the other by @sid-6581 in https://github.com/neovide/neovide/pull/2527
* fix: Use composition to fix window transparency when using D3D by @fredizzimo in https://github.com/neovide/neovide/pull/2529
* chore: add cfg to is_tty so that it isn't built on windows by @Kethku in https://github.com/neovide/neovide/pull/2537
* fix: dont resize the d3d buffers if the window size is zero by @Kethku in https://github.com/neovide/neovide/pull/2538

## New Contributors
* @gmr458 made their first contribution in https://github.com/neovide/neovide/pull/2331
* @hhirtz made their first contribution in https://github.com/neovide/neovide/pull/2334
* @DZappala made their first contribution in https://github.com/neovide/neovide/pull/2374
* @emonadeo made their first contribution in https://github.com/neovide/neovide/pull/2351
* @polachok made their first contribution in https://github.com/neovide/neovide/pull/2323
* @nazriel made their first contribution in https://github.com/neovide/neovide/pull/2441
* @Theaninova made their first contribution in https://github.com/neovide/neovide/pull/2478
* @Holzhaus made their first contribution in https://github.com/neovide/neovide/pull/2256
* @InfyniteHeap made their first contribution in https://github.com/neovide/neovide/pull/2490
* @cpmsmith made their first contribution in https://github.com/neovide/neovide/pull/2545

**Full Changelog**: https://github.com/neovide/neovide/compare/0.12.2...0.13.0

<hr /><em>This discussion was created from the release <a href='https://github.com/neovide/neovide/releases/tag/0.13.0'>0.13.0</a>.</em>

**Comment by chrisgrieser** [comment]
Big thank you for 0.13 – it solves a lot of issues on macOS 🥳

**Comment by PriceHiller** [comment]
Looking awesome, thanks for Neovide and thanks to everyone else invovled 🙂

**Comment by kamoshi** [comment]
I upgraded today and while Neovide was already pretty good, this new version feels even smoother and nicer to use!

Thanks for the hard work :heart:
